server/channels/app/platform/shared_channel_notifier.go

```diff
@@ -14,16 +14,16 @@ import (
 	"github.com/mattermost/mattermost/server/v8/platform/services/sharedchannel"
 )
 
-var sharedChannelEventsForSync model.StringArray = []string{
-	model.WebsocketEventPosted,
-	model.WebsocketEventPostEdited,
-	model.WebsocketEventPostDeleted,
-	model.WebsocketEventReactionAdded,
-	model.WebsocketEventReactionRemoved,
+var sharedChannelEventsForSync model.WebsocketEventTypeArray = []model.WebsocketEventType{
+	model.Posted,
+	model.PostEdited,
+	model.PostDeleted,
+	model.ReactionAdded,
+	model.ReactionRemoved,
 }
```

Comment:
I don't think we need a separate `WebsocketEventTypeArray`. A `[]model.WebsocketEventType` should be enough here. `isEligibleForEvents` can use `utils.Contains` to check if the even is part of the slice.

Reply:
fair point, have removed the type, although will need the `ToStringArray` function to cast it to string array and use `utils.Contains`